code/jobs/CleanupJob.php

```diff
@@ -70,52 +80,56 @@ public function getTitle() {
 	 * Set immediacy of job
 	 * @return int
 	 */
-	public function getJobType() {
+	public function getJobType()
+	{
 		$this->totalSteps = '1';
 		return QueuedJob::IMMEDIATE;
 	}
 
 	/**
 	 * Clear out stale jobs based on the cleanup values
 	 */
-	public function process() {
+	public function process()
+	{
 		$statusList = implode('\', \'', $this->config()->cleanup_statuses);
-		switch($this->config()->cleanup_method) {
-			// If Age, we need to get jobs that are at least n days old
+		switch ($this->config()->cleanup_method) {
+            // If Age, we need to get jobs that are at least n days old
 			case "age":
-				$cutOff = date("Y-m-d H:i:s",
+				$cutOff = date(
+					"Y-m-d H:i:s",
 					strtotime(SS_Datetime::now() .
 						" - " .
 						$this->config()->cleanup_value .
-						" days"
-					)
+						" days")
 				);
 				$stale = DB::query(
 					'SELECT "ID" 
-					FROM "QueuedJobDescriptor" 
-					WHERE "JobStatus" 
-					IN (\'' . $statusList . '\')
-					AND "LastEdited" < \'' . $cutOff .'\''
+                    FROM "QueuedJobDescriptor" 
+                    WHERE "JobStatus" 
+                    IN (\'' . $statusList . '\') 
+                    AND "LastEdited" < \'' . $cutOff . '\' 
+                    LIMIT ' . $this->config()->query_limit
```

Comment:
You could probably extract this into a variable and assert it's a valid integer first.

```php
$limit = '';
$configuration = $this->config()->get('query_limit');
if (is_numeric($configuration)) {
	$limit = 'LIMIT ' . ((int) $configuration);
}
```

Reply:
@ScopeyNZ Agreed, and I'd briefly considered that but wound up following the existing patterns. I've overwritten the prievious commit with this change (and because my IDE had wrecked havok on the formatting).